ouroboros-network/src/Ouroboros/Network/TxSubmission/Inbound.hs

```diff
@@ -63,7 +65,16 @@ data TxSubmissionMempoolWriter txid tx idx m =
        mempoolAddTxs :: [tx] -> m [txid]
     }
 
-data TraceTxSubmissionInbound txid tx = TraceTxSubmissionInbound --TODO
```

Comment:
We avoid partial fields (I'm pretty sure that's as true for the Network Layer as it is from Consensus; @coot?)

You have a few options, but here's a relatively simple one.

```
data TraceTxSubmissionInbound txid tx =
      -- | Transactions just about to be inserted.
      TraceTxSubmissionCollected !Int
      -- | Just accepted this many transactions.
    | TraceTxSubmissionAccepted !Int
      -- | Just rejected this many transactions.
    | TraceTxSubmissionRejected !Int
```

You'd have two `traceWiths` back-to-back after the `mempoolAddTxs` call, which is a bit weird but not a giant problem. If you want to avoid that, I'd recommend defining a small single-constructor record type and passing that as thes ole argument to the merged `*Processed` event.

Reply:
This will cost has a non-zero runtime cost, but it's probably small and excellent idea for avoiding the warning.

Reply:
Done!

Reply:
@newhoggy, @nfrisby, I think in this case we could just drop the field names to avoid partial fields.

I'm really conscious of causing two traces in place of one.

Reply:
> I'm really conscious of causing two traces in place of one.

I sympathize with that. I called it "weird but not a giant problem", but it does seem sloppy to me. I've assumed it wouldn't have a significant cost in a full node, but maybe my intuition is off for eg benchmarking use cases, maybe it adds up here and there, etc.

> I think in this case we could just drop the field names to avoid partial fields.

I would push back on `TraceTxSubmissionProcessed !Int !Int` -- that seems like a food gun to mix up the two `Int`s. So I think it would be worthwhile to introduce the

```
data ProcessedTxCount = ProcessedTxCount {
      ptcAccepted :: !Int
    , ptcRejected :: !Int
    }
```

Reply:
> So I think it would be worthwhile to introduce the separate type

:+1: on that!

Reply:
Done.  Thanks!